chainpy/eth/managers/contracthandler.py

```diff
@@ -116,130 +134,126 @@ def get_contract_by_addr(self, contract_addr: EthAddress) -> Optional[EthContrac
                 return contract
         return None
 
-    def get_contract_by_event_name(self, event_name: str) -> Optional[EthContract]:
-        contract_name = self._contract_name_by_event_name.get(event_name)
-        return self.get_contract_by_name(contract_name)
+    def get_contracts_by_event_name(self, event_name: str) -> List[EthContract]:
+        data = self._event_db.get(event_name)
+        if data is None:
+            return list()
+        return [data["contract"] for data in self._event_db[event_name]]
```

Comment:
l.138에서 이미 `data`로 `self._event_db[event_name]`를 꺼냈으니 또 꺼낼 필요 없을 것 같습니다
```suggestion
        return [event["contract"] for event in data]
```